packages/react-form/src/hooks/list/reducer.ts

```diff
@@ -21,6 +23,16 @@ interface ReinitializeAction<Item> {
   payload: {list: Item[]};
 }
 
+interface AddFieldsAction<Item> {
+  type: 'addFields';
+  payload: {list: Item[]};
+}
+
+interface RemoveFieldsAction {
```

Comment:
It might be better to use a singular form `RemoveFieldAction` as it only removes one field at a time.